### DISCUSSION TITLE
when i compile deepin-unioncode  error,some head file can't find

### DISCUSSION THREAD

**Discussion by wufanqq** [linuxdeepin/developer-center#9552] [Q&A | 问答板块]
I encountered several issues while attempting to compile deepin-unioncode following the steps in the README:

When executing sudo apt build-dep ./:
The following packages have unmet dependencies:
builddeps:./ : Depends: libdtkgui-dev but it is not installable
                   Depends: libdtkcore5-bin but it is not installable
After completing the CMake process, during compilation, I encounter errors stating that the header files for Dwidget, DguiApplicationHelper, DabstractDialog, and Dmenu cannot be found.
I am unsure how to resolve these problems. Could you please assist?

我在按照readme中的步骤进行编译deepin-unioncode时遇到几个问题如下：
1、sudo apt build-dep ./ ：
      The following packages have unmet dependencies:
      builddeps:./ : Depends: libdtkgui-dev but it is not installable
                           Depends: libdtkcore5-bin but it is not installable
2、在完成cmake后进行编译，报错：
      找不到Dwidget DguiapllicationHelper DabstractDialog Dmenu 这些头文件
 不知道该如何解决？

**Comment by BLumia** [comment]
> The following packages have unmet dependencies:
> builddeps:./ : Depends: libdtkgui-dev but it is not installable
> Depends: libdtkcore5-bin but it is not installable

你必然需要先解决此问题。由于你没提供你所使用的系统信息，这里假定你在使用 deepin v23，则建议先更新系统，然后重试此步骤。

> 2、在完成cmake后进行编译，报错：
> 找不到Dwidget DguiapllicationHelper DabstractDialog Dmenu 这些头文件

这是由于忽视了上面的错误导致的。请先解决上述第一个问题。
